src/supermarket/config/initializers/session_store.rb

```diff
@@ -2,4 +2,5 @@
 
 Rails.application.config.session_store :cookie_store,
                                        key: "_supermarket_session",
-                                       secure: Supermarket::Host.secure_session_cookie?
+                                       secure: Rails.env.production? && \
```

Comment:
Any Specific reason for checking in production.

Reply:
Yea, because in local we dont have enough setup to get through this check `                                               Supermarket::Host.secure_session_cookie?` and it does not let the application create a session in local for that reason, a csrf error is always reported since we are using gem `gem "omniauth-rails_csrf_protection"`. Only way we could make the application work in local as of now is by commenting out this gem but after upgrading omniauth to 2.0.0 it makes sure that the csrf protection check is there and does not continue authentication if not present . In local it was impossible to check functionality without a session creation to store authentication token